### ISSUE TITLE
PKCE support?

### ISSUE THREAD

**Issue by bmerry** [CZ-NIC/pyoidc#551]
This is more a question than a bug report. Does pyoidc support PKCE (RFC 7636), and if so, how does one use it? I went digging into the code and found [this](https://github.com/OpenIDC/pyoidc/blob/69a7ef59c1f40c29313354e034bc15bfff92921b/src/oic/oic/__init__.py#L660), which seems to handle adding the code challenge to the authorization request, but the code verifier doesn't seem to be saved anywhere to send with the access token request.

**Comment by tpazderka** [comment]
Yes, it looks like a partial implenmentation. The storing of the code_verifier is not handled in the code.

**Comment by Goldziher** [comment]
I assume this issue is stale? If so I will work to add a PR for this in the near future.

**Comment by schlenk** [comment]
A PR would be welcome.

**Comment by bmerry** [comment]
We ended up not using OpenID, so this is no longer relevant to me and I don't mind whether it is kept open for a future PR or closed due to my question being answered.

**Comment by tpazderka** [comment]
I don't mind keeping this open as a TODO.

**Comment by Goldziher** [comment]
I'm afraid our client requires this, so I will implement this and add a PR.

**Comment by Goldziher** [comment]
Well, project cancelled. No go.
